rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/services/ServiceConfig.java

```diff
@@ -51,19 +56,43 @@ RetrySupport retrySupport() {
   @Bean
   ClouddriverService clouddriverService(
       Ok3Client ok3Client, RequestInterceptor spinnakerRequestInterceptor) {
-    ObjectMapper objectMapper =
-        new ObjectMapper()
-            .enable(DeserializationFeature.READ_UNKNOWN_ENUM_VALUES_AS_NULL)
-            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
 
     return new RestAdapter.Builder()
         .setEndpoint(clouddriverBaseUrl)
         .setRequestInterceptor(spinnakerRequestInterceptor)
         .setClient(ok3Client)
-        .setConverter(new JacksonConverter(objectMapper))
+        .setConverter(new JacksonConverter(getObjectMapper()))
         .setLogLevel(RestAdapter.LogLevel.valueOf(retrofitLogLevel))
         .setErrorHandler(SpinnakerRetrofitErrorHandler.getInstance())
         .build()
         .create(ClouddriverService.class);
   }
+
+  private ObjectMapper getObjectMapper() {
+    return new ObjectMapper()
+        .enable(DeserializationFeature.READ_UNKNOWN_ENUM_VALUES_AS_NULL)
+        .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+  }
+
+  /*As part of retrofit2 changes, creating clouddriverservice with retrofit2 API changes */
+  @Bean
+  ClouddriverRetrofit2Service clouddriverRetrofit2Service(
+      OkHttp3ClientConfiguration okHttpClientConfig) {
+
+    OkHttpClient.Builder okHttpClientBuilder = okHttpClientConfig.createForRetrofit2();
+    okHttpClientBuilder.addInterceptor(
```

Comment:
Does it make sense to move this to createForRetrofit2?

Reply:
I don't think any intermediate interceptor got into place, I will move it to createForRetrofit2 itself